### DISCUSSION TITLE
patch的翻譯

### DISCUSSION THREAD

**Discussion by ken71301** [python/python-docs-zh-tw#591] [Add Term]
### 初步確認

- [X] 我已經確認過 [術語列表](https://github.com/python/python-docs-zh-tw/wiki/%E8%A1%93%E8%AA%9E%E5%88%97%E8%A1%A8) 中沒有相關術語。
- [X] 我已經確認過 [Python 官方術語列表](https://docs.python.org/zh-tw/3/glossary.html) 中沒有相關術語。

### 術語原文

patch / patching

### 術語翻譯

修補 或 模擬

### 說明

出處：unittest.mock.po

原因：若單純以語意與樂詞網的詞彙翻譯，patch應翻譯為「修補」，但在mock相關的文件中，patch的意義較接近於mock本身的「模擬」
例如：Additionally, mock provides a [patch()](https://docs.python.org/3.12/library/unittest.mock.html#unittest.mock.patch) decorator that handles patching module and class level attributes within the scope of a test......

因此想提出針對patch翻譯的討論，究竟該如何翻譯較為正確

**Comment by cschan1828** [comment]
查了幾本權威字典，patch 這個詞本身應該沒有「模擬」的意思，不過在 Mock 模組的文件中顯然也不是「修補」，硬要說的話是製造一個 shadow 去「替換」真的物件（有點像補丁那樣，明顯與周圍其他部份不同）。

這邊我會建議用原文 patch，不翻譯。

**Comment by mattwang44** [comment]
被搶先惹XD

單指 unittest.mock.patch 的翻譯的話，我會傾向不翻譯、譯文直接用 patch，不然實在不太好想像是指那個 patch。
另外，patch 與 mock 我覺得意思還是不太一樣，mock 是指用來模仿真實行為的一個 object，patch 則是指將真實物件用 mock 替換掉的動作。
